pkg/oci/oci.go

```diff
@@ -109,15 +110,23 @@ func GetGadgetImage(ctx context.Context, image string, authOpts *AuthOptions, pu
 		}
 	}
 
-	manifest, err := getImageManifestForArch(ctx, imageStore, image, authOpts)
+	manifest, err := getImageManifest(ctx, imageStore, image, authOpts)
 	if err != nil {
 		return nil, fmt.Errorf("getting arch manifest: %w", err)
 	}
 
-	prog, err := getEbpfProgramFromManifest(ctx, imageStore, manifest)
+	prog, err := getLayerFromManifest(ctx, imageStore, manifest, eBPFObjectMediaType)
 	if err != nil {
 		return nil, fmt.Errorf("getting ebpf program: %w", err)
 	}
+	if len(prog) == 0 {
```

Comment:
This works since `len(prog) == 0` if `prog = nil`. But this empty check is already done in `getLayerFromManifest`.

I prefer to write `prog == nil` with an error message "no ebpf program found" or "layer"